### ISSUE TITLE
replace/rename `toJson` to something else

### ISSUE THREAD

**Issue by chirastefan** [MichaelMarner/dart-redux-remote-devtools#38]
Is there a way to replace or rename this method?
Context: I'm using https://pub.dev/packages/dart_mappable which generates `toJson`(returns `String`) and `toMap`(returns `Map<String,dynamic>`) on a decorated class. What I need from that class is `toMap` which is what `dart-redux-remote-devtools` expects `toJson` to return;

**Comment by MichaelMarner** [comment]
This package uses `toJson` because that is what [jsonEncode](https://api.flutter.dev/flutter/dart-convert/jsonEncode.html) expects.

But yes, you can provide your own encoder when initialising remote devtools. Something like:

```dart
final remoteDevtools = RemoteDevToolsMiddleware('192.168.1.52:8000', 
  stateEncoder: (dynamic state) => state.toMap(),
);
```

**Comment by chirastefan** [comment]
<img width="786" alt="Screenshot 2023-05-06 at 23 20 49" src="https://user-images.githubusercontent.com/16649178/236645133-1e8a8180-1973-467b-b134-ce86d80d0c32.png">
and it doesn't seem to work

**Comment by MichaelMarner** [comment]
Sorry, I'm not familiar with dart mappables package so I think I gave you the wrong code snippet.

State Encoder needs to return a JSON sting. The default implementation is to use Dart's built in `jsonEncode`, which tries to use a `toJson` method on any objects it needs to encode.

As you are overriding the state encoder, you need to return a string. This actually sounds like you need to use Dart Mappables' toJson method:

```dart
final remoteDevtools = RemoteDevToolsMiddleware('192.168.1.52:8000', 
  stateEncoder: (dynamic state) => state.toJson(),
);
```

**Comment by chirastefan** [comment]
yes, but I don't think that's the issue I have with this. I'm not sure that `stateEncoder` is the solution for me.
Before I used `json_serializable` for which I had to create `toJson` and `fromJson` methods that point to the generated methods from the generated file and those functions return a Map<String, dynamic>.
This library `dart_mappable` generates `toJson`(returns String) and `toMap`(returns Map<String, dynamic>) - this is the method I need because I want to see the state obj not a long string in the dev tools.
Is there a way to tell redux_dev_tools to look for `toMap` instead of `toJson` or any other solution for this issue?
@MichaelMarner
